test/jdk/java/math/BigDecimal/SquareRootTests.java

```diff
@@ -171,6 +172,77 @@ private static int compareSqrtImplementations(BigDecimal bd, MathContext mc) {
                                 bd.sqrt(mc), "sqrt(" + bd + ") under " + mc);
     }
 
+    private static int exactResultTests() {
+        int failures = 0;
+        MathContext unnecessary = new MathContext(1, RoundingMode.UNNECESSARY);
+        MathContext arbitrary = new MathContext(0, RoundingMode.CEILING);
+
+        BigDecimal[] errCases = {
+                // (strippedScale & 1) != 0
+                BigDecimal.TEN,
+                // (strippedScale & 1) == 0 && !stripped.isPowerOfTen() && sqrtRem[1].signum != 0
+                BigDecimal.TWO,
+        };
+
+        for (BigDecimal input : errCases) {
+            BigDecimal result;
+            // mc.roundingMode == RoundingMode.UNNECESSARY
+            try {
+                result = input.sqrt(unnecessary);
+                System.err.println("Unexpected sqrt with UNNECESSARY RoundingMode: (" + input + ").sqrt() = " + result);
+                failures += 1;
+            } catch (ArithmeticException e) {
+                // Expected
+            }
+
+            // mc.roundingMode != RoundingMode.UNNECESSARY && mc.precision == 0
+            try {
+                result = BigDecimal.TEN.sqrt(arbitrary);
```

Comment:
```suggestion
                result = input.sqrt(arbitrary);
```

Reply:
For the other paths, looking at already existing tests, it seems to me that they should already cover all the possible paths for every `RoundingMode`. There remains only the case when `workingScale` overflows, but I'm not sure if it requires a test, as future implementations could extend the range of the values supported by the algorithm...

Reply:
> OK. But for the sake of completeness, I would add at least one test case for an overflowing `workingScale`.

I'm afraid it's not possible to produce such a test case, with the current implementation of `BigInteger`.
Indeed, `workingScale` is defined by `workingScale = this.scale - normScale`, and `normScale` by:
```java
long normScale = minWorkingPrec - this.precision() + this.scale;
normScale += normScale & 1L;
```
Therefore `workingScale ==  this.precision() - minWorkingPrec - (normScale & 1L)`, so to overflow `workingScale` it is necessary to maximize `this.precision()`, thus`this.intVal` must have at least `Integer.MAX_VALUE + 1` digits, but `BigInteger.TEN.pow(Integer.MAX_VALUE)` surely the supported range for  `BigInteger`s...

Reply:
Actually, this proves also that `workingScale` should never overflow...

Reply:
I don't get your point. Here's an example:
```
BigDecimal.ONE.sqrt(new MathContext(2_000_000_000, RoundingMode.UP))
|  Exception java.lang.ArithmeticException: Overflow
|        at BigDecimal.sqrt (BigDecimal.java:2226)
|        at (#4:1)
```

Reply:
> I don't get your point. Here's an example:
> 
> ```
> BigDecimal.ONE.sqrt(new MathContext(2_000_000_000, RoundingMode.UP))
> |  Exception java.lang.ArithmeticException: Overflow
> |        at BigDecimal.sqrt (BigDecimal.java:2226)
> |        at (#4:1)
> ```

Yeas, that's correct indeed, I was wrongly assuming that `this.precision() >= minWorkingPrec`.

Reply:
This might be due to the special treatment of powers of 10 radicands in the current implementation.
I think we can forget about special cases, so I'd say that there's no need to reconsider your implementation.

Before approving, however, I'll check with the original implementor.

Reply:
@fabioromano1 Let's wait for @jddarcy to chime in next week about the issue with the special handling of 10^(2e) powers with huge requested precision.

In the meantime, please replace 2022 with 2024 in the copyright line of the test class (L.2), and add the issue number 8341402 on L.26.

I hope we can integrate before [Rampdown Phase One](https://openjdk.org/projects/jdk/24/).